packages/relay/tests/lib/precheck.spec.ts

```diff
@@ -247,6 +247,31 @@ describe('Precheck', async function () {
       expect(result).to.not.exist;
     });
 
+    it('should recognize if a signed raw transaction is the deterministic deployment transaction', async () => {
+      const parsedDeterministicDeploymentTransaction = ethers.Transaction.from(
+        constants.DETERMINISTIC_DEPLOYER_TRANSACTION,
+      );
+
+      expect(Precheck.isDeterministicDeploymentTransaction(parsedDeterministicDeploymentTransaction)).to.be.true;
+    });
+
+    it('Should recognize if a signed raw transaction is NOT the deterministic deployment transaction', async () => {
+      expect(Precheck.isDeterministicDeploymentTransaction(parsedtxWithChainId0x0)).to.be.false;
+      expect(Precheck.isDeterministicDeploymentTransaction(parsedTxWithMatchingChainId)).to.be.false;
+      expect(Precheck.isDeterministicDeploymentTransaction(parsedTxWithNonMatchingChainId)).to.be.false;
+    });
+
+    it('should pass for gas price if the transaction is the deterministic deployment transaction', async function () {
+      const parsedDeterministicDeploymentTransaction = ethers.Transaction.from(
+        constants.DETERMINISTIC_DEPLOYER_TRANSACTION,
+      );
+      const result = precheck.gasPrice(
+        parsedDeterministicDeploymentTransaction,
+        100 * constants.TINYBAR_TO_WEIBAR_COEF,
```

Comment:
would this test be more meaningful of the gasPrice were a low number like 0 or 1 tinyBar?

Reply:
Hmm my understanding for this case is this. The `100 * constants.TINYBAR_TO_WEIBAR_COEF` (100 hbars) arg right there is the minimum gasPrice to compared against the `parsedDeterministicDeploymentTransaction.gasPrice` (10 hbars). 

The comparing condition is
```
const passes = txGasPrice >= minGasPrice || Precheck.isDeterministicDeploymentTransaction(tx);
```
in which, `txGasPrice = parsedDeterministicDeploymentTransaction.gasPrice` (10 hbars), and `minGasPrice = 100 * constants.TINYBAR_TO_WEIBAR_COEF` (100 hbars).

So if the minGasPrice is set to 0 or 1, then the condition `txGasPrice >= minGasPrice` (10 >= 1) will be true, and regardless the outcome of the `Precheck.isDeterministicDeploymentTransaction(tx);` this `passes` will be true. 

But if the minGasPrice is set to be greater than 10, then the condition `txGasPrice >= minGasPrice` (10 >= 100) will be false -> now this condition will rely on the outcome of the `Precheck.isDeterministicDeploymentTransaction(tx);` function -> and because it's the deterministic deployment the outcome will be true -> the `passes` will be true.

More, isn't the whole purpose of this feature is to bypass deterministic deployment when its gasPrice (10hbar) is too low compared to  network's gasPrice (minimum is 71 hbars for localnet)?

Reply:
Ah makes sense.  I was incorrect in the way that the parameter was used.